### ISSUE TITLE
Editor keyboard navigation does not work correctly when the tools are specifically configured

### ISSUE THREAD

**Issue by martintabakov** [telerik/kendo-ui-core#7718]
### Bug report

When the tools configuration of the Editor is set, the formatting tool becomes active when focused, and trying to navigate to the previous/next tool using the arrow keys results in changing the selected item in the formatting tool.

### Reproduction of the problem

1. Open this Dojo example - https://dojo.telerik.com/ImIhIHuh
2. Click in the textarea
3. Press Shift + Tab - the Formatting tool gets focused and is already active
4. Press the right arrow key

### Current behavior

The Formatting tool comes active when focused. As a result, pressing the arrow key results in change of the selected item in the Formatting tool

### Expected/desired behavior

The formatting tool shouldn't be active when focused, and arrow key press should navigate to the next tool

### Environment

* **Kendo UI version:** 2024.1.130
* **Browser:** [all]
